<H1>BACHELOR OF SCIENCE IN INFORMATION TECHNOLOGY<H1>

<H1>STUDENT ANALYSIS : MARK DAVE L RODRIGUEZ<H1>

In [2]:
import numpy as np
import pandas as pd
import warnings
warnings.filterwarnings('ignore')

In [3]:


uk_road_accident = pd.read_csv(r"C:\Users\CLIENT\Downloads\uk_road_accident.csv")
uk_road_accident


Index Accident_Severity Accident Date   Latitude  \
0       200701BS64157           Serious      5/6/2019  51.506187   
1       200701BS65737           Serious      2/7/2019  51.495029   
2       200701BS66127           Serious    26-08-2019  51.517715   
3       200701BS66128           Serious    16-08-2019  51.495478   
4       200701BS66837            Slight      3/9/2019  51.488576   
...               ...               ...           ...        ...   
660674  201091NM01760            Slight    18-02-2022  57.374005   
660675  201091NM01881            Slight    21-02-2022  57.232273   
660676  201091NM01935            Slight    23-02-2022  57.585044   
660677  201091NM01964           Serious    23-02-2022  57.214898   
660678  201091NM02142           Serious    28-02-2022  57.575210   

                   Light_Conditions           District Area  Longitude  \
0             Darkness - lights lit  Kensington and Chelsea  -0.209082   
1                          Daylight  Kensington and Chelsea  -0.173647   
2       Darkness - lighting unknown  Kensington and Chelsea  -0.210215   
3                          Daylight  Kensington and Chelsea  -0.202731   
4             Darkness - lights lit  Kensington and Chelsea  -0.192487   
...                             ...                     ...        ...   
660674                     Daylight                Highland  -3.467828   
660675       Darkness - no lighting                Highland  -3.809281   
660676                     Daylight                Highland  -3.862727   
660677       Darkness - no lighting                Highland  -3.823997   
660678                     Daylight                Highland  -3.895673   

        Number_of_Casualties  Number_of_Vehicles Road_Surface_Conditions  \
0                          1                   2                     Dry   
1                          1                   2             Wet or damp   
2                          1                   3                     Dry   
3                          1                   4                     Dry   
4                          1                   2                     Dry   
...                      ...                 ...                     ...   
660674                     2                   1                     Dry   
660675                     1                   1            Frost or ice   
660676                     1                   3            Frost or ice   
660677                     1                   2             Wet or damp   
660678                     1                   1             Wet or damp   

                 Road_Type Urban_or_Rural_Area     Weather_Conditions  \
0       Single carriageway               Urban     Fine no high winds   
1       Single carriageway               Urban  Raining no high winds   
2                      NaN               Urban                    NaN   
3       Single carriageway               Urban     Fine no high winds   
4                      NaN               Urban                    NaN   
...                    ...                 ...                    ...   
660674  Single carriageway               Rural     Fine no high winds   
660675  Single carriageway               Rural     Fine no high winds   
660676  Single carriageway               Rural     Fine no high winds   
660677  Single carriageway               Rural     Fine no high winds   
660678    Dual carriageway               Rural  Snowing no high winds   

                                Vehicle_Type  
0                                        Car  
1                                        Car  
2                      Taxi/Private hire car  
3       Bus or coach (17 or more pass seats)  
4                              Other vehicle  
...                                      ...  
660674                                   Car  
660675                                   Car  
660676                                   Car  
660677                 Motorcycle over 500cc  
660

<h1>1. Average number of casualties per accident<h1>

In [4]:
print("Average number of casualties per accident:", uk_road_accident["Number_of_Casualties"].mean())

Average number of casualties per accident: 1.357040257068864


<h1>INSIGHT : The result shows the typical number of casualties per accident, giving an idea of how severe accidents generally are in the dataset.<h1>

<H1>2. Average number of vehicles per accident<H1>

In [5]:
print("Average number of vehicles per accident:", uk_road_accident["Number_of_Vehicles"].mean())

Average number of vehicles per accident: 1.8312554205597575


<h1>Insight: The average number of vehicles per accident shows whether crashes usually involve one vehicle or multiple vehicles.</h1>

<H1>3.Most frequent Accident Severity<H1>

In [6]:
print("Most frequent Accident Severity:", uk_road_accident["Accident_Severity"].value_counts())

Most frequent Accident Severity: Accident_Severity
Slight     563801
Serious     88217
Fatal        8661
Name: count, dtype: int64


<h1>Insight : The most frequent accident severity reveals if accidents are mostly slight, serious, or fatal.</h1>

<H1>4.Least frequent Accident Severity<H1>

In [7]:
print("Q4: Least frequent Accident Severity:", uk_road_accident["Accident_Severity"].value_counts())

Q4: Least frequent Accident Severity: Accident_Severity
Slight     563801
Serious     88217
Fatal        8661
Name: count, dtype: int64


<h1>Insight : The least frequent accident severity shows the rarest crash outcome, helping to identify uncommon risks.</h1>

<H1>5 . Most common Weather Condition<H1>

In [8]:
print("Most common Weather Condition:\n\n",uk_road_accident["Weather_Conditions"].value_counts())

Most common Weather Condition:

 Weather_Conditions
Fine no high winds       520885
Raining no high winds     79696
Other                     17150
Raining + high winds       9615
Fine + high winds          8554
Snowing no high winds      6238
Fog or mist                3528
Snowing + high winds        885
Name: count, dtype: int64


<h1>Insight : The most common weather condition during accidents highlights how climate affects road safety.</h1>

<h1>6. Least common Weather Condition<h1>

In [9]:
print("Least common Weather Condition:\n\n", uk_road_accident["Weather_Conditions"].value_counts())

Least common Weather Condition:

 Weather_Conditions
Fine no high winds       520885
Raining no high winds     79696
Other                     17150
Raining + high winds       9615
Fine + high winds          8554
Snowing no high winds      6238
Fog or mist                3528
Snowing + high winds        885
Name: count, dtype: int64


<h1>Insight : The least common weather condition shows when accidents are least likely to happen.</h1>

<h1>7.  Most frequent Road Surface Condition<h1>

In [10]:
print("Most frequent Road Surface Condition:", uk_road_accident["Road_Surface_Conditions"].value_counts())

Most frequent Road Surface Condition: Road_Surface_Conditions
Dry                     447821
Wet or damp             186708
Frost or ice             18517
Snow                      5890
Flood over 3cm. deep      1017
Name: count, dtype: int64


<h1>Insight : The most frequent road surface condition indicates which surface (dry, wet, icy) is most associated with crashes.</h1>

<h1>8. Least frequent Road Surface Condition<h1>

In [11]:
print("Least frequent Road Surface Condition:\n\n", uk_road_accident["Road_Surface_Conditions"].value_counts())

Least frequent Road Surface Condition:

 Road_Surface_Conditions
Dry                     447821
Wet or damp             186708
Frost or ice             18517
Snow                      5890
Flood over 3cm. deep      1017
Name: count, dtype: int64


<h1>Insight : The least frequent road surface condition reveals where accidents are least likely to occur.</h1>

<h1> 9. Most common Road Type <h1>

In [12]:
print("Most common Road Type:", uk_road_accident["Road_Type"].value_counts())

Most common Road Type: Road_Type
Single carriageway    492143
Dual carriageway       99424
Roundabout             43992
One way street         13559
Slip road               7041
Name: count, dtype: int64


<h1>Insight : The most common road type highlights whether crashes happen more often on motorways, single carriageways, or other roads.</h1>

<h1>10. Least common Road Type<h1>

In [13]:
print("Least common Road Type:", uk_road_accident["Road_Type"].value_counts())

Least common Road Type: Road_Type
Single carriageway    492143
Dual carriageway       99424
Roundabout             43992
One way street         13559
Slip road               7041
Name: count, dtype: int64


<h1>Insight : The least common road type shows which roads experience fewer accidents.</h1>

<h1>11. Most frequent Light Condition<h1>

In [14]:
print("Most frequent Light Condition:", uk_road_accident["Light_Conditions"].value_counts())

Most frequent Light Condition: Light_Conditions
Daylight                       484880
Darkness - lights lit          129335
Darkness - no lighting          37437
Darkness - lighting unknown      6484
Darkness - lights unlit          2543
Name: count, dtype: int64


<h1>Insight : The most frequent light condition shows whether crashes happen more in daylight, darkness, or poor lighting.</h1>

<h1>12 . Least frequent Light Condition<h1>

In [15]:
print("Q12: Least frequent Light Condition:", uk_road_accident["Light_Conditions"].value_counts())

Q12: Least frequent Light Condition: Light_Conditions
Daylight                       484880
Darkness - lights lit          129335
Darkness - no lighting          37437
Darkness - lighting unknown      6484
Darkness - lights unlit          2543
Name: count, dtype: int64


<h1>Insight : The least frequent light condition indicates the lighting environment least associated with crashes.</h1>

<h1>13 .  Average accident location latitude and longitude<h1>

In [16]:
print("LATITUDE :")
print("Average accident latitude:", uk_road_accident["Latitude"].mean())
print("LONGITUDE :")
print("Average accident longitude:", uk_road_accident["Longitude"].mean())

LATITUDE :
Average accident latitude: 52.553865761110956
LONGITUDE :
Average accident longitude: -1.4312103685020727


<h1>Insight: The average accident location is around latitude 52.55 and longitude -1.43, meaning most accidents occur near the central region of the UK.</h1>


<H1>14 . Area (Urban/Rural) with most accidents<H1>

In [17]:
print("Area with most accidents:", uk_road_accident["Urban_or_Rural_Area"].value_counts())

Area with most accidents: Urban_or_Rural_Area
Urban          421663
Rural          238990
Unallocated        11
Name: count, dtype: int64


<h1>Insight : The area with the most accidents urban or rural highlights where road incidents are concentrated.</h1>

<H1>15. Area (Urban/Rural) with least accidents<H1>

In [18]:
print("Area with least accidents:", uk_road_accident["Urban_or_Rural_Area"].value_counts())

Area with least accidents: Urban_or_Rural_Area
Urban          421663
Rural          238990
Unallocated        11
Name: count, dtype: int64


<h1>Insight Q13: The area with the least accidents shows where crashes happen less often.</h1>

<H1>16 .Most frequent vehicle type<H1>

In [19]:
print("Most frequent vehicle type:", uk_road_accident["Vehicle_Type"].value_counts())

Most frequent vehicle type: Vehicle_Type
Car                                      497992
Van / Goods 3.5 tonnes mgw or under       34160
Bus or coach (17 or more pass seats)      25878
Motorcycle over 500cc                     25657
Goods 7.5 tonnes mgw and over             17307
Motorcycle 125cc and under                15269
Taxi/Private hire car                     13294
Motorcycle over 125cc and up to 500cc      7656
Motorcycle 50cc and under                  7603
Goods over 3.5t. and under 7.5t            6096
Other vehicle                              5637
Minibus (8 - 16 passenger seats)           1976
Agricultural vehicle                       1947
Pedal cycle                                 197
Data missing or out of range                  6
Ridden horse                                  4
Name: count, dtype: int64


<h1>Insight : The most frequent vehicle type in accidents shows which vehicles are most at risk on the road.</h1>

<H1>17. Average number of casualties for each Accident Severity category<H1>

In [20]:

print(uk_road_accident.groupby("Accident_Severity")["Number_of_Casualties"].mean())

Accident_Severity
Fatal      1.903129
Serious    1.467280
Slight     1.331402
Name: Number_of_Casualties, dtype: float64


<h1>Insight : The average number of casualties per accident severity reveals how seriousness affects the number of people injured.</h1>

<H1>18 .  District Area with most accidents<H1>

In [21]:
print("Q18: District Area with most accidents:", uk_road_accident["District Area"].value_counts())

Q18: District Area with most accidents: District Area
Birmingham            13491
Leeds                  8898
Manchester             6720
Bradford               6212
Sheffield              5710
                      ...  
Berwick-upon-Tweed      153
Teesdale                142
Shetland Islands        133
Orkney Islands          117
Clackmannanshire         91
Name: count, Length: 422, dtype: int64


<h1>Insight : The district area with the most accidents shows which locality faces the greatest road safety challenge.</h1>

<H1>19.District Area with least accidents<H1>

In [22]:
print(" District Area with least accidents:", uk_road_accident["District Area"].value_counts())

 District Area with least accidents: District Area
Birmingham            13491
Leeds                  8898
Manchester             6720
Bradford               6212
Sheffield              5710
                      ...  
Berwick-upon-Tweed      153
Teesdale                142
Shetland Islands        133
Orkney Islands          117
Clackmannanshire         91
Name: count, Length: 422, dtype: int64


<h1>Insight : The district area with the least accidents shows where crashes are rarest.</h1>

<H1>20 .Least frequent vehicle type<H1>

In [23]:
print("Least frequent vehicle type:", uk_road_accident["Vehicle_Type"].value_counts())

Least frequent vehicle type: Vehicle_Type
Car                                      497992
Van / Goods 3.5 tonnes mgw or under       34160
Bus or coach (17 or more pass seats)      25878
Motorcycle over 500cc                     25657
Goods 7.5 tonnes mgw and over             17307
Motorcycle 125cc and under                15269
Taxi/Private hire car                     13294
Motorcycle over 125cc and up to 500cc      7656
Motorcycle 50cc and under                  7603
Goods over 3.5t. and under 7.5t            6096
Other vehicle                              5637
Minibus (8 - 16 passenger seats)           1976
Agricultural vehicle                       1947
Pedal cycle                                 197
Data missing or out of range                  6
Ridden horse                                  4
Name: count, dtype: int64


<h1>Insight : The least frequent vehicle type shows which vehicles are least likely to be involved in crashes.</h1>

In [24]:
uk_road_accident['Accident Date'] = pd.to_datetime(uk_road_accident['Accident Date'], dayfirst=True, errors= 'coerce')

In [25]:
uk_road_accident.isnull().sum()

Index                           0
Accident_Severity               0
Accident Date              395672
Latitude                       25
Light_Conditions                0
District Area                   0
Longitude                      26
Number_of_Casualties            0
Number_of_Vehicles              0
Road_Surface_Conditions       726
Road_Type                    4520
Urban_or_Rural_Area            15
Weather_Conditions          14128
Vehicle_Type                    0
dtype: int64

In [26]:
uk_road_accident['Year'] = uk_road_accident['Accident Date'].dt.year
uk_road_accident['Month'] = uk_road_accident['Accident Date'].dt.month
uk_road_accident['Day'] = uk_road_accident['Accident Date'].dt.day
uk_road_accident['DayOfWeek'] = uk_road_accident['Accident Date'].dt.dayofweek

<h1>21. In 2019, how many accidents occurred in Urban and Rural areas?<h1>

In [33]:
uk_road_accident[uk_road_accident['Year'] == 2019]['Urban_or_Rural_Area'].value_counts()



Urban_or_Rural_Area
Urban          45253
Rural          26611
Unallocated        2
Name: count, dtype: int64

<h1>INSIGHTS : The result shows if accidents were more common in crowded urban roads or in rural areas with higher speeds and fewer safety measures.<h1>

<h1>22. Which year recorded the most accidents?<h1>

In [34]:
uk_road_accident['Year'].value_counts()

Year
2019.0    71867
2020.0    70163
2021.0    66172
2022.0    56805
Name: count, dtype: int64

<h1>Insight: This shows the accident that happened per year, helping identify which year had the highest risk on the roads.<h1>

<h1>23. Which month had the most accidents overall?<h1>

In [35]:
uk_road_accident['Month'].value_counts()

Month
11.0    24240
12.0    24156
10.0    23962
7.0     22939
9.0     22558
2.0     22264
6.0     22196
3.0     21824
5.0     21723
8.0     21106
4.0     19787
1.0     18252
Name: count, dtype: int64

<h1>Insight: The result highlights which month accidents peaked, possibly due to weather, holidays, or seasonal travel.<h1>

<h1>24. Which day of the week in 2020 had the most accidents?<h1>

In [36]:
uk_road_accident[uk_road_accident['Year'] == 2020]['DayOfWeek'].value_counts()


DayOfWeek
5.0    11655
2.0    11412
3.0    10626
1.0     9765
6.0     9711
4.0     9698
0.0     7296
Name: count, dtype: int64

<h1>Insight:

Accidents occurred most often on Saturdays, while Mondays had the fewest, suggesting weekend travel increases accident risks compared to weekday commutes.<h1>

<h1>25 . How many accidents happened per month in 2020?<h1>


In [37]:
uk_road_accident[uk_road_accident['Year'] == 2020]['Month'].value_counts()


Month
12.0    6727
10.0    6275
7.0     6215
2.0     6022
11.0    5872
9.0     5869
5.0     5837
6.0     5770
3.0     5579
4.0     5457
8.0     5394
1.0     5146
Name: count, dtype: int64

<h1>Insight:

The result shows which months in 2019 had the highest number of accidents, indicating possible seasonal effects such as holiday travel or weather conditions.
<h1>